### PR TITLE
Models added

### PR DESCRIPTION
**Pull request by tpardeshi**
Models has been added.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,6 +9,7 @@ gem 'puma', '~> 3.11'
 gem 'jbuilder', '~> 2.7'
 gem 'bootsnap', '>= 1.4.2', require: false
 gem 'mysql2', '>= 0.5.2'
+gem 'pg'
```
Comment (rahulbajaj0509) [inline Gemfile]: We already have a `mysql2` database, I don't think we require `pg` here.

File: app/models/product.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Product < ApplicationRecord
+
```
Comment (rahulbajaj0509) [inline app/models/product.rb]: Nitpick: 
a) I can see 4 space indentation, lets keep it to two.
b) Extra line.

File: app/models/locale.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Locale < ApplicationRecord
+	has_many :screenshots
+	belongs_to :products
+
```
Comment (rahulbajaj0509) [inline app/models/locale.rb]: Nitpick: Extra line.

File: db/migrate/20191025064042_create_products.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+class CreateProducts < ActiveRecord::Migration[6.0]
+ def up
+
```
Comment (rahulbajaj0509) [inline db/migrate/20191025064042_create_products.rb]: Extra spaces and lines.

File: db/migrate/20191025064435_create_versions.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+class CreateVersions < ActiveRecord::Migration[6.0]
+
```
Comment (rahulbajaj0509) [inline db/migrate/20191025064435_create_versions.rb]: extra spaces and lines.

File: db/migrate/20191025064714_create_locales.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  def up
+    create_table :locales do |t|
+    	t.string :name
+    	t.timestamps
+    end
+  end
+
+def down
+	drop_table :locales
+
```
Comment (rahulbajaj0509) [inline db/migrate/20191025064714_create_locales.rb]: extra line.
Add a line at the EOF.

File: db/migrate/20191025065957_create_screenshots.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+class CreateScreenshots < ActiveRecord::Migration[6.0]
+  def up
+    create_table :screenshots do |t|
+		t.string :name
+    	t.timestamps
+    end
+  end
+
+  def down
+
```
Comment (rahulbajaj0509) [inline db/migrate/20191025065957_create_screenshots.rb]: indentation

File: db/migrate/20191025070223_add_foreign_key_details.rb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+class AddForeignKeyDetails < ActiveRecord::Migration[6.0]
+ def up
+
```
Comment (rahulbajaj0509) [inline db/migrate/20191025070223_add_foreign_key_details.rb]: same

**Comment by rahulbajaj0509** [comment]
Also, please squash the commits once you are done with the required changes.

**Comment by rahulbajaj0509** [comment]
@pravins can you help Twinkle a bit with git?

File: db/migrate/20191025064714_create_locales.rb [hunk 9]
```diff
@@ -1,13 +1,11 @@
 class CreateLocales < ActiveRecord::Migration[6.0]
   def up
-    create_table :locales do |t|
+	create_table :locales do |t|
```
Comment (rahulbajaj0509) [inline db/migrate/20191025064714_create_locales.rb]: Same

File: db/schema.rb [hunk 10]
```diff
@@ -12,24 +12,21 @@
 
 ActiveRecord::Schema.define(version: 2019_10_25_070223) do
 
-  # These are extensions that must be enabled in order to support this database
-  enable_extension "plpgsql"
-
-  create_table "locales", force: :cascade do |t|
+  create_table "locales", options: "ENGINE=InnoDB DEFAULT CHARSET=latin1", force: :cascade do |t|
```
Comment (rahulbajaj0509) [inline db/schema.rb]: Can you please add this file to the .gitignore?

Comment (tpardeshi) [reply db/schema.rb]: Sure.

Comment (rahulbajaj0509) [reply db/schema.rb]: Also, can you please squash the commits?

Comment (pravins) [reply db/schema.rb]: Right, we missed doing squash yesterday.

File: Gemfile [hunk 11]
```diff
@@ -4,12 +4,11 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 ruby '2.6.3'
 
 gem 'rails', '~> 6.0.0'
-gem 'sqlite3', '~> 1.4'
+gem 'mysql2', '>= 0.5.2'
```
[the remark this answers is not in the document]
Comment (tpardeshi) [reply Gemfile]: I used "rails db:system:change --to=mysql" for database change from pg to mysql. so there is a change in the line.

File: db/migrate/20191025064042_create_products.rb [hunk 12]
```diff
@@ -1,11 +1,11 @@
...
-	create_table :products do |t|
-		t.string :name
-		t.timestamps
-	end
-  end
-  def down
-	drop_table :products
-  end
+	def up
+		create_table :products do |t|
```
Comment (rahulbajaj0509) [inline db/migrate/20191025064042_create_products.rb]: Indentation.

File: db/migrate/20191025064714_create_locales.rb [hunk 13]
```diff
@@ -1,11 +1,11 @@
...
-    	t.timestamps
-    end
-  end
-  def down
-	drop_table :locales
-  end
+	def up
+		create_table :locales do |t|
+    		t.string :name
+    		t.timestamps
```
Comment (rahulbajaj0509) [inline db/migrate/20191025064714_create_locales.rb]: Same.

**Comment by tpardeshi** [comment]
@rahulbajaj0509, Done with squashed commit (1a814d5d0ee6fdcbc9979e0eb802b77a93d91c2b). Models now has said fixes. Please have a look.

Thanks.

**Comment by rahulbajaj0509** [comment]
I can still see 7 commits for this PR instead of one.

**Comment by rahulbajaj0509** [comment]
90% of them will go away if you rebase and squash the commits

**Comment by tpardeshi** [comment]
@rahulbajaj0509, resubmitted squashed commit for Models. Looking forward ...

**Comment by rahulbajaj0509** [comment]
I don't think adding 4 models should add arounf 5k lines of code. Can you please add unwanted files to the `.gitignore` file

File: db/migrate/20191025064042_create_products.rb [hunk 14]
```diff
@@ -0,0 +1,12 @@
+class CreateProducts < ActiveRecord::Migration[6.0]
+  def up
+    create_table :products do |t|
+      t.string :name
+      t.timestamps
+	  end
+  end
+
+  def down
+	  drop_table :products
```
Comment (rahulbajaj0509) [inline db/migrate/20191025064042_create_products.rb]: come on, don't make me comment for indentation any more!

File: db/migrate/20191025064714_create_locales.rb [hunk 15]
```diff
@@ -0,0 +1,11 @@
+class CreateLocales < ActiveRecord::Migration[6.0]
+  def up
+	  create_table :locales do |t|
```
Comment (rahulbajaj0509) [inline db/migrate/20191025064714_create_locales.rb]: same

File: db/migrate/20191025065957_create_screenshots.rb [hunk 16]
```diff
@@ -0,0 +1,11 @@
+class CreateScreenshots < ActiveRecord::Migration[6.0]
+  def up
+    create_table :screenshots do |t|
+      t.string :name
+      t.timestamps
+    end
+  end
+  def down
```
Comment (rahulbajaj0509) [inline db/migrate/20191025065957_create_screenshots.rb]: Please install rubocop on your development environment!

Comment (tpardeshi) [reply db/migrate/20191025065957_create_screenshots.rb]: Already installed.

**Comment by rahulbajaj0509** [comment]
Please squash your commits. Also, please add the vendor/bin folder in your .gitignre file.

If this PR does not have the expected changes till next week, I will close this PR and create one myself and merge. Sorry, but we need to move a little faster on this.

**Comment by tpardeshi** [comment]
Changes have been already pushed to the PR previous week. I am resolving some of the Rubocop offenses which are almost done. Will be doing squash commit by today.

**Comment by rahulbajaj0509** [comment]
In general this PR should not exceed more than 100 lines.

**Comment by rahulbajaj0509** [comment]
Any reason for not rebasing?

**Comment by tpardeshi** [comment]
Done with rebase and squash of commits.

**Comment by rahulbajaj0509** [comment]
@sourabh-upadhyay can you please review this PR. @tpardeshi can you add the model structure to the PR that I shared with you(in the form of an image)?

**Comment by tpardeshi** [comment]
@rahulbajaj0509  Yes done.

File: app/models/version.rb [hunk 17]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+# The Version class represents the version model
+class Version < ApplicationRecord
+  has_many :screenshots
+  belongs_to :products
```
Comment (sourabh-upadhyay) [inline app/models/version.rb]: `belongs_to` must be singular

For example: [rails guide](https://guides.rubyonrails.org/association_basics.html#the-belongs-to-association)

`belongs_to :product`

File: app/models/screenshot.rb [hunk 18]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+# The Screenshot class represents the screenshot model
+class Screenshot < ApplicationRecord
+  belongs_to :locales
```
Comment (sourabh-upadhyay) [inline app/models/screenshot.rb]: Follow the same comment https://github.com/lingostack/lingoqa/pull/2/files?file-filters%5B%5D=.png&file-filters%5B%5D=.rb&hide-deleted-files=true#r350658393

**Comment by sourabh-upadhyay** [comment]
> @sourabh-upadhyay can you please review this PR. @tpardeshi can you add the model structure to the PR that I shared with you(in the form of an image)?

The code seems good. A couple of small comments based on the Rails convention for `model` while defining association.

**Comment by tpardeshi** [comment]
@sourabh-upadhyay  @rahulbajaj0509 
Thanks for the review. I will do the changes.

**Comment by sourabh-upadhyay** [comment]
> @sourabh-upadhyay @rahulbajaj0509
> Thanks for the review. I will do the changes.

LGTM! Good work.

**Comment by rahulbajaj0509** [comment]
@tpardeshi thanks, merging!

File: .gitignore [hunk 19]
```diff
@@ -7,6 +7,8 @@
 /.bundle
 /db/*.sqlite3
 /db/*.sqlite3-journal
+/db/schema.rb
+db/seeds.rb
```
Comment (rahulbajaj0509) [inline .gitignore]: Just noticed, these files should be in .gitignore
Will add a commit to remove it from here. Don't worry about it.

**Pull request merged**